File: 0908. Smallest Range I/0908.java
// time complexity: O(nlogn)
// space complexity: O(1)
class Solution {
    public int smallestRangeI(int[] nums, int k) {
        Arrays.sort(nums);
        int min = nums[0] + k; // add k to the smallest
        int max = nums[nums.length-1] - k; // substract k from the largest
        // if min is smaller than max, return the difference
        return max <= min ? 0 : max - min;
    }
}

File: 0908. Smallest Range I/0908.py
# time complexity: O(n)
# space complexity: O(1)
class Solution(object):
    def smallestRangeI(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        # initialize min and max
        low = float('inf')
        high = float('-inf')

        # find min and max in one traversal
        for num in nums:
            if num < low:
                low = num
            if num > high:
                high = num

        # compute the minimized range
        diff = (high-k) - (low+k)

        # if diff is negative, score is 0
        return diff if diff > 0 else 0